ci: windows short tests database error

Truncated test results:

```
ERROR test/test_scanner.py::TestScanner::test_version_in_package[https://downloads.openwrt.org/releases/packages-19.07/x86_64/packages/-znc_1.7.4-1_x86_64.ipk-znc-1.7.4-other_products916] - sqlite3.OperationalError: unable to open database file
ERROR test/test_scanner.py::TestScanner::test_version_in_package[http://mirror.centos.org/centos/8/BaseOS/x86_64/os/Packages/-zsh-5.5.1-6.el8_1.2.x86_64.rpm-zsh-5.5.1-other_products917] - sqlite3.OperationalError: unable to open database file
ERROR test/test_scanner.py::TestScanner::test_version_in_package[https://ftp.netbsd.org/pub/pkgsrc/packages/NetBSD/amd64/9.1/All/-zsh-5.8nb2.tgz-zsh-5.8-other_products918] - sqlite3.OperationalError: unable to open database file
ERROR test/test_scanner.py::TestScanner::test_version_in_package[http://ftp.br.debian.org/debian/pool/main/z/zsh/-zsh_5.8-6+b1_amd64.deb-zsh-5.8-other_products919] - sqlite3.OperationalError: unable to open database file
ERROR test/test_scanner.py::TestScanner::test_does_not_scan_symlinks - sqlite3.OperationalError: unable to open database file
ERROR test/test_scanner.py::TestScanner::test_cannot_open_file - sqlite3.OperationalError: unable to open database file
ERROR test/test_scanner.py::TestScanner::test_clean_file_path - sqlite3.OperationalError: unable to open database file
= 29 failed, 493 passed, 55 skipped, 21 warnings, 1043 errors in 233.96s (0:03:53) =
```

I'm not sure what's up here.  It *could* be related to our switch to using `--offline` on the windows short tests, but I can't figure out why there'd only be 29 failures if there was a problem with the database.  It should have failed at least 270 times for all the test_version_in_package tests if it couldn't read the database at all.  So I'm guessing it's something else, but what?

Github won't let me paste even the "short test summary" version but here's the top few tests that fail in case that's relevant to why the database stops responding:

```python
2023-03-01T20:48:34.1234193Z =========================== short test summary info ===========================
2023-03-01T20:48:34.1234959Z FAILED test/test_checkers.py::TestCheckerVersionParser::test_filename_is[sqlite-sqlite3-expected_results44] - sqlite3.OperationalError: unable to open database file
2023-03-01T20:48:34.1235459Z FAILED test/test_csv2cve.py::TestCSV2CVE::test_csv2cve_valid_file - RuntimeError: This event loop is already running
2023-03-01T20:48:34.1236357Z FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[True-exploits_list0-product_info0-triage_info0-CRITICAL-EXPLOIT] - sqlite3.OperationalError: no such table: cve_range
2023-03-01T20:48:34.1238408Z FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[False-exploits_list1-product_info1-triage_info1-CRITICAL] - sqlite3.OperationalError: no such table: cve_range
2023-03-01T20:48:34.1239231Z FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[True-exploits_list2-product_info2-triage_info2-CRITICAL] - sqlite3.OperationalError: no such table: cve_range
2023-03-01T20:48:34.1239885Z FAILED test/test_exploits.py::TestExploitScanner::test_exploit_checker[False-exploits_list3-product_info3-triage_info3-CRITICAL] - sqlite3.OperationalError: no such table: cve_range
2023-03-01T20:48:34.1240986Z FAILED 
```

I'm wondering if maybe one of the tests is clobbering the cve_range table or locking the database somehow.

--- a/.github/workflows/cve_scan.yml
+++ b/.github/workflows/cve_scan.yml
@@ -24,8 +24,8 @@ jobs:
       - name: Get cached database
         uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
-          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+          path: cache
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - name: Install dependencies and cve-bin-tool
         run: |
           python -m pip install --upgrade pip
@@ -33,6 +33,7 @@ jobs:
           python -m pip install --upgrade wheel
           python -m pip install --upgrade pytest
           pip install . -r doc/requirements.txt
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
       - name: Test to check for CVEs for Python requirements and HTML report dependencies
         run: |
           pytest test/test_requirements.py

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -62,7 +62,7 @@ jobs:
       - name: Get cached database
         uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
+          path: cache
           key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - name: Install cabextract
         run: sudo apt-get update && sudo apt-get install cabextract
@@ -79,7 +79,9 @@ jobs:
           python -m pip install --upgrade .
       - name: Try single CLI run of tool
         run: |
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
           NO_EXIT_CVE_NUM=1 python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out -n json
+          cp -r ~/.cache/cve-bin-tool cache
       - name: Run async tests
         run: >
           pytest -n 4 -v
@@ -112,7 +114,7 @@ jobs:
       - name: Get cached database
         uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
+          path: cache
           key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - uses: technote-space/get-diff-action@v6
         with:
@@ -145,7 +147,9 @@ jobs:
           python -m pip install --editable .
       - name: Try single CLI run of tool
         run: |
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
           NO_EXIT_CVE_NUM=1 python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out -n json
+          cp -r ~/.cache/cve-bin-tool cache
       - name: Run async tests
         env:
           LONG_TESTS: ${{ steps.git-diff.outputs.value }}
@@ -189,7 +193,7 @@ jobs:
       - name: Get cached database
         uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
+          path: cache
           key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - uses: technote-space/get-diff-action@v6
         with:
@@ -225,7 +229,9 @@ jobs:
           python -m playwright install chromium --with-deps
       - name: Try single CLI run of tool
         run: |
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
           NO_EXIT_CVE_NUM=1 python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out -n json
+          cp -r ~/.cache/cve-bin-tool cache
       - name: Run all tests which rely on external connectivity
         env:
           EXTERNAL_SYSTEM: ${{ steps.git-diff.outputs.value }}
@@ -260,8 +266,14 @@ jobs:
       - name: Get cached database
         uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
-          key: linux-cve-bin-tool-${{ steps.get-date.outputs.DATE }}
+          path: cache
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.DATE }}
+          enableCrossOsArchive: true
+          fail-on-cache-miss: true
+      - name: Move cache to ~/.cache/cve-bin-tool
+        run: |
+          mkdir '~\.cache'
+          mv cache '~\.cache\cve-bin-tool'
       - name: Install cve-bin-tool
         run: |
           python -m pip install --upgrade pip
@@ -271,7 +283,7 @@ jobs:
           python -m pip install --upgrade .
       - name: Try single CLI run of tool
         run: |
-          python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out --offline -n json 
+          python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out --offline -n json
       - name: Run async tests
         run: >
           pytest -n 4 -v
@@ -306,8 +318,13 @@ jobs:
       - name: Get cached database
         uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
-          key: linux-cve-bin-tool-${{ steps.get-date.outputs.DATE }}
+          path: cache
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.DATE }}
+          enableCrossOsArchive: true
+      - name: Move cache to ~/.cache/cve-bin-tool
+        run: |
+          mkdir '~\.cache'
+          mv cache '~\.cache\cve-bin-tool'
       - name: Install cve-bin-tool
         run: |
           python -m pip install --upgrade pip
@@ -362,4 +379,3 @@ jobs:
           flags: win-longtests
           name: codecov-umbrella
           fail_ci_if_error: false
-

--- a/.github/workflows/testing.yml
+++ b/.github/workflows/testing.yml
@@ -63,7 +63,7 @@ jobs:
         uses: actions/cache@v3
         with:
           path: cache
-          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - name: Install cabextract
         run: sudo apt-get update && sudo apt-get install cabextract
       - name: Install OS dependencies for testing PDF
@@ -115,7 +115,7 @@ jobs:
         uses: actions/cache@v3
         with:
           path: cache
-          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - uses: technote-space/get-diff-action@v6
         with:
           PATTERNS: |
@@ -194,7 +194,7 @@ jobs:
         uses: actions/cache@v3
         with:
           path: cache
-          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - uses: technote-space/get-diff-action@v6
         with:
           PATTERNS: |

--- a/.github/workflows/update-cache.yml
+++ b/.github/workflows/update-cache.yml
@@ -29,8 +29,8 @@ jobs:
           echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
       - uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
-          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+          path: cache
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - name: Install cve-bin-tool
         run: |
           python -m pip install --upgrade pip
@@ -39,6 +39,8 @@ jobs:
           python -m pip install --editable .
       - name: Update database
         run: |
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
           python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out -u now
+          cp -r ~/.cache/cve-bin-tool cache
 
 # Windows cache is currently disabled.

--- a/doc/how_to_guides/cve_scanner_gh_action.md
+++ b/doc/how_to_guides/cve_scanner_gh_action.md
@@ -74,7 +74,7 @@ jobs:
       # as exit code. And GitHub terminates action when process produces
       # nonzero exit code status.
       - name: Upload report as an artifact
-        # his will upload generated report as an GitHub artifact which you can download later.
+        # This will upload generated report as an GitHub artifact which you can download later.
         uses: actions/upload-artifact@v2
         with:
           name: cve_report

--- a/doc/how_to_guides/cve_scanner_gh_action.md
+++ b/doc/how_to_guides/cve_scanner_gh_action.md
@@ -1,4 +1,3 @@
-
 # How do I run CVE Binary Tool in GitHub Actions?
 
 CVE Binary Tool can be run in GitHub Actions in order to scan your project dependencies for any CVEs. To understand what GitHub Actions are and how they work, you can refer to the [official documentation](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions).
@@ -14,66 +13,68 @@ To add the tool to your workflow, place the [`cve_scanner_gh_action.yml`](https:
 ```yml
 name: CVE scanner
 on:
-#    You can customize this according to your need.
+  # You can customize this according to your need.
   - push
   - pull_request
 jobs:
   build_and_scan:
     runs-on: ubuntu-latest
     steps:
-    # Get date utility for caching database.
+      # Get date utility for caching database.
       - name: Get Date
         id: get-date
         run: |
           echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
         shell: bash
-    #  Let's first download dependencies for this action.
+      # Let's first download dependencies for this action.
       - uses: actions/checkout@v2
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
           python-version: '3.x'
-    #  This second step is unnecessary but highly recommended because
-    #  It will cache database and saves time re-downloading it if database isn't stale.
+      # This second step is unnecessary but highly recommended because
+      # It will cache database and saves time re-downloading it if database isn't stale.
       - name: get cached python packages
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           path: ~/.cache/pip
           key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
           restore-keys: |
             ${{ runner.os }}-pip-
       - name: get cached database
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
-          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+          path: cache
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - name: Install CVE Binary Tool
-#          We are using latest development version of CVE Binary Tool
-#          because current PyPI version don't have features like config file support,
-#          generating HTML report etc.
-        run: pip install git+https://github.com/intel/cve-bin-tool@master
-#          In case you prefer current PyPI version, you need to hard code CLI options
-#          for cve-bin-tool in the action itself and have to use CSV or JSON as output format.
-#          pip install cve-bin-tool
+        # We are using latest development version of CVE Binary Tool
+        # because current PyPI version don't have features like config file support,
+        # generating HTML report etc.
+        run: |
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
+          pip install git+https://github.com/intel/cve-bin-tool@master
+      # In case you prefer current PyPI version, you need to hard code CLI options
+      # for cve-bin-tool in the action itself and have to use CSV or JSON as output format.
+      # pip install cve-bin-tool
       - name: build package
-#          Here, we are building Python wheel for this example.
-#          You need to replace this with your build process.
+        # Here, we are building Python wheel for this example.
+        # You need to replace this with your build process.
         run: |
           pip install wheel
           python setup.py bdist_wheel
       - name: Scan built package
-#          Here, we are scanning built wheel which is situated in /dist directory
-#          Python stores built packages in /dist directory.
-#          You need to replace it with the directory where you have stored built package
+        # Here, we are scanning built wheel which is situated in /dist directory
+        # Python stores built packages in /dist directory.
+        # You need to replace it with the directory where you have stored built package
         run: cve-bin-tool dist -f html -o cve-bin-tool-report.html -x
-#          Alternatively if you have written config file for cve-bin-tool you can use following command
-#          cve-bin-tool -C path/to/cve_bin_tool_config.toml
+        #  Alternatively if you have written config file for cve-bin-tool you can use following command
+        #  cve-bin-tool -C path/to/cve_bin_tool_config.toml
         continue-on-error: true
-#          You need to set continue_on_error: true because CVE Binary Tool sets number of cves
-#          as exit code. And GitHub terminates action when process produces
-#          nonzero exit code status.
+      # You need to set continue_on_error: true because CVE Binary Tool sets number of cves
+      # as exit code. And GitHub terminates action when process produces
+      # nonzero exit code status.
       - name: Upload report as an artifact
-#        This will upload generated report as an GitHub artifact which you can download later.
+        # his will upload generated report as an GitHub artifact which you can download later.
         uses: actions/upload-artifact@v2
         with:
           name: cve_report

--- a/doc/how_to_guides/cve_scanner_gh_action.yml
+++ b/doc/how_to_guides/cve_scanner_gh_action.yml
@@ -1,65 +1,67 @@
 name: CVE scanner
 on:
-#    You can customize this according to your need.
+  # You can customize this according to your need.
   - push
   - pull_request
 jobs:
   build_and_scan:
     runs-on: ubuntu-latest
     steps:
-    # Get date utility for caching database.
+      # Get date utility for caching database.
       - name: Get Date
         id: get-date
         run: |
           echo "date=$(/bin/date -u "+%Y%m%d")" >> $GITHUB_OUTPUT
         shell: bash
-    #  Let's first download dependencies for this action.
+      # Let's first download dependencies for this action.
       - uses: actions/checkout@v2
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
           python-version: '3.x'
-    #  This second step is unnecessary but highly recommended because
-    #  It will cache database and saves time redownloading it if database isn't stale.
+      # This second step is unnecessary but highly recommended because
+      # It will cache database and saves time redownloading it if database isn't stale.
       - name: get cached python packages
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
           path: ~/.cache/pip
           key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
           restore-keys: |
             ${{ runner.os }}-pip-
       - name: get cached database
-        uses: actions/cache@v2
+        uses: actions/cache@v3
         with:
-          path: ~/.cache/cve-bin-tool
-          key: ${{ runner.os }}-cve-bin-tool-${{ steps.get-date.outputs.date }}
+          path: cache
+          key: Linux-cve-bin-tool-${{ steps.get-date.outputs.date }}
       - name: Install CVE Binary Tool
-#          We are using latest development version of CVE Binary Tool
-#          because current PyPI version don't have features like config file support,
-#          generating HTML report etc.
-        run: pip install git+https://github.com/intel/cve-bin-tool@master
-#          In case you prefer current PyPI version, you need to hard code CLI options
-#          for cve-bin-tool in the action itself and have to use CSV or JSON as output format.
-#          pip install cve-bin-tool
+        # We are using latest development version of CVE Binary Tool
+        # because current PyPI version don't have features like config file support,
+        # generating HTML report etc.
+        run: |
+          [[ -e cache ]] && mkdir -p .cache && mv cache ~/.cache/cve-bin-tool
+          pip install git+https://github.com/intel/cve-bin-tool@master
+      # In case you prefer current PyPI version, you need to hard code CLI options
+      # for cve-bin-tool in the action itself and have to use CSV or JSON as output format.
+      # pip install cve-bin-tool
       - name: build package
-#          Here, we are building Python wheel for this example.
-#          You need to replace this with your build process.
+        # Here, we are building Python wheel for this example.
+        # You need to replace this with your build process.
         run: |
           pip install wheel
           python setup.py bdist_wheel
       - name: Scan built package
-#          Here, we are scanning built wheel which is situated in /dist directory
-#          Python stores built packages in /dist directory.
-#          You need to replace it with the directory where you have stored built package
+        # Here, we are scanning built wheel which is situated in /dist directory
+        # Python stores built packages in /dist directory.
+        # You need to replace it with the directory where you have stored built package
         run: cve-bin-tool dist -f html -o cve-bin-tool-report.html -x
-#          Alternatively if you have written config file for cve-bin-tool you can use following command
-#          cve-bin-tool -C path/to/cve_bin_tool_config.toml
+        # Alternatively if you have written config file for cve-bin-tool you can use following command
+        # cve-bin-tool -C path/to/cve_bin_tool_config.toml
         continue-on-error: true
-#          You need to set continue_on_error: true because CVE Binary Tool sets number of cves
-#          as exit code. And GitHub terminates action when process produces
-#          nonzero exit code status.
+      # You need to set continue_on_error: true because CVE Binary Tool sets number of cves
+      # as exit code. And GitHub terminates action when process produces
+      # nonzero exit code status.
       - name: Upload report as an artifact
-#        This will upload generated report as an GitHub artifact which you can download later.
+        # This will upload generated report as an GitHub artifact which you can download later.
         uses: actions/upload-artifact@v2
         with:
           name: cve_report

